[EC2SpotManager] AssertionError: Instance with EC2 ID ... is not in our database.

```
[2017-11-01 20:30:38,000] [INFO] [start_monitoring_daemon]: [Pool 117] Needs to be cycled, terminating all instances...
[2017-11-01 20:30:39,416] [INFO] [start_monitoring_daemon]: [Pool 117] Termination complete.
[2017-11-01 20:31:03,117] [WARNING] [start_monitoring_daemon]: [Pool 11] No allowed region was cheap enough to spawn instances.
[2017-11-01 20:31:23,023] [INFO] [start_monitoring_daemon]: [Pool 117] Deleting terminated instance with EC2 ID i-097dc49e3962b28b0 from our database.
[2017-11-01 20:31:23,030] [INFO] [start_monitoring_daemon]: [Pool 117] Deleting terminated instance with EC2 ID i-02907cf62f8a906fc from our database.
[2017-11-01 20:31:23,037] [INFO] [start_monitoring_daemon]: [Pool 117] Deleting terminated instance with EC2 ID i-0edb9f7f337972dad from our database.
[2017-11-01 20:31:23,044] [INFO] [start_monitoring_daemon]: [Pool 117] Deleting terminated instance with EC2 ID i-0fcf36a9124777609 from our database.
[2017-11-01 20:31:23,051] [INFO] [start_monitoring_daemon]: [Pool 117] Deleting terminated instance with EC2 ID i-0ffba9c3ce9057752 from our database.
[2017-11-01 20:31:23,060] [INFO] [start_monitoring_daemon]: [Pool 117] Deleting terminated instance with EC2 ID i-0a6660c939f16b25f from our database.
[2017-11-01 20:31:23,067] [INFO] [start_monitoring_daemon]: [Pool 117] Deleting terminated instance with EC2 ID i-0fb16cc7fce5f6df3 from our database.
[2017-11-01 20:31:23,073] [INFO] [start_monitoring_daemon]: [Pool 117] Deleting terminated instance with EC2 ID i-082a2743ee3586a13 from our database.
[2017-11-01 20:31:23,079] [INFO] [start_monitoring_daemon]: [Pool 117] Deleting terminated instance with EC2 ID i-077f119733d61dc1b from our database.
[2017-11-01 20:31:23,085] [INFO] [start_monitoring_daemon]: [Pool 117] Deleting terminated instance with EC2 ID i-0db9a0a676a000639 from our database.
[2017-11-01 20:31:23,093] [INFO] [start_monitoring_daemon]: [Pool 117] Deleting terminated instance with EC2 ID i-0529f7ca855678047 from our database.
[2017-11-01 20:31:23,100] [INFO] [start_monitoring_daemon]: [Pool 117] Deleting terminated instance with EC2 ID i-0dc14b48d44a641ba from our database.
[2017-11-01 20:31:23,107] [INFO] [start_monitoring_daemon]: [Pool 117] Needs 12 more instances, starting...
[2017-11-01 20:31:38,372] [INFO] [start_monitoring_daemon]: [Pool 117] Creating 12 instances...
[2017-11-01 20:31:56,338] [INFO] [start_monitoring_daemon]: [Pool 117] Successfully created 12 instances, 0 requests timed out and were canceled
[2017-11-01 20:32:25,010] [INFO] [start_monitoring_daemon]: [Pool 117] Deleting instance with EC2 ID i-05acca957016374ba from our database, has no corresponding machine on EC2.
[2017-11-01 20:32:25,010] [INFO] [start_monitoring_daemon]: [Pool 117] Deleting instance with EC2 ID i-05acca957016374ba from our database.
[2017-11-01 20:32:25,020] [INFO] [start_monitoring_daemon]: [Pool 117] Needs 1 more instances, starting...
[2017-11-01 20:32:41,881] [INFO] [start_monitoring_daemon]: [Pool 117] Creating 1 instances...
[2017-11-01 20:32:53,961] [INFO] [start_monitoring_daemon]: [Pool 117] Successfully created 1 instances, 0 requests timed out and were canceled
[2017-11-01 20:33:26,857] [ERROR] [start_monitoring_daemon]: [Pool 117] Instance with EC2 ID i-05acca957016374ba is not in our database.
```

There was no ec2spotmanager web interface activity at the time.

--- a/server/ec2spotmanager/tasks.py
+++ b/server/ec2spotmanager/tasks.py
@@ -312,6 +312,7 @@ def update_requests(provider, region, pool_id):
 
             for req_id in successful_requests:
                 instance = requested[req_id]
+                instance.created = timezone.now()  # reset creation time now that the instance really exists
                 instance.hostname = successful_requests[req_id]['hostname']
                 instance.instance_id = successful_requests[req_id]['instance_id']
                 instance.status_code = successful_requests[req_id]['status_code']
@@ -446,6 +447,11 @@ def update_instances(provider, region):
             reasons = []
 
             if instance.instance_id not in debug_cloud_instance_ids_seen:
+                if timezone.now() - instance.created < timezone.timedelta(minutes=5):
+                    logger.warning("[Pool %d] Machine %s has no corresponding machine in %s/%s, "
+                                   "not deleting until it is at least 5 minutes old.",
+                                   instance.pool_id, instance.instance_id, instance.provider, instance.region)
+                    continue
                 reasons.append("no corresponding machine on cloud")
 
             if instance.instance_id in debug_not_updatable_continue:

